### PR TITLE
RFE: Add plugin name field into plugin context use it to improve error reporting

### PR DESCRIPTION
**Pull request by rpv-tomsk**
Superseedes #2011 
Superseedes #2799

This PR is the next attempt to implement this enhancement.
Unlike previous, this one does not try to touch each plugin. 
Instead, only `configfile.c` functions are affected.

I think this is a shortest way to reach results.

cc @rubenk 

Ruben, what do you think?

### CODE REVIEW

**Comment by rpv-tomsk** [comment]
TODO: look into `src/daemon/common.c`: `service_name_to_port_number()`, `parse_value()` and so on.

**Comment by rpv-tomsk** [comment]
cc  @rubenk 

Ruben, what do you think about this PR in overall?
About functions/macro naming?

Is this good for go in ?

File: src/daemon/plugin.h [hunk 1]
```diff
@@ -398,6 +399,15 @@ int parse_notif_severity(const char *severity);
 #define DEBUG(...) /* noop */
 #endif             /* ! COLLECT_DEBUG */
 
+/* These will log messages, prefixed by plugin name */
```
Comment (rubenk) [inline src/daemon/plugin.h]: s/These/This/

**Comment by rubenk** [comment]
I left one minor remark, but otherwise LGTM!

**Pull request merged**